pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -510,10 +520,12 @@ func (s *Controller) WorkloadInstanceHandler(wi *model.WorkloadInstance, event m
 		s.serviceInstances.deleteInstances(key, instancesDeleted)
 	}
 
-	if event == model.EventDelete {
-		s.serviceInstances.deleteInstances(key, instances)
-	} else {
-		s.serviceInstances.updateInstances(key, instances)
+	if len(instances) > 0 {
```

Comment:
nit: IMO this is not needed, both calls will do nothing if there are no instances already.

Since there is no diff its just a style thing, but IMO its still nice to call them. Should be same cost and it would allow the functions to evolve to do something with just the key, for example

Reply:
I donot think we need to change here, unless you want to do something else